### DISCUSSION TITLE
Change the CFBundleExecutable on macOS

### DISCUSSION THREAD

**Discussion by streamingdv** [fvarrui/JavaPackager#438] [Q&A]
The reason why I'm asking this is because when my application is requesting mic access for example (or any other permission) the request permission dialog looks like as shown in the attached image
<img width="270" alt="Rename" src="https://github.com/user-attachments/assets/e9c594ec-8cb5-4e34-915a-cbb5b298c5f0">
Is there a way to change that?

**Comment by fvarrui** [comment]
Hi @streamingdv! 
What are your entitlements?

**Reply by streamingdv** [reply]
My macConfig looks like that at the moment

```
        String macStartupType
        if (arch != null && arch.contains("aarch64")) {
            macStartupType = 'ARM64'
        } else {
            macStartupType = 'X86_64'
        }

        macConfig {
            appId = 'com.grill.psplay.PXPlay'
            icnsFile = file('src/main/resources/com/grill/psplay/image/PXPlay.icns')
            generateDmg = true
            generatePkg = false
            relocateJar = true
            macStartup = macStartupType

            codesignApp = false

            infoPlist = new io.github.fvarrui.javapackager.model.InfoPlist()
            infoPlist.bundlePackageType = 'APPL'
            infoPlist.additionalEntries = '''
        <key>NSMicrophoneUsageDescription</key>
        <string>For mic support, PXPlay requires access to your microphone.</string>
        <key>LSApplicationCategoryType</key>
        <string>public.app-category.games</string>
    '''
        }
```

**Reply by streamingdv** [reply]
@fvarrui I found out that there is a property in the macOS build called customLauncher by looking at the source code. I just copied the existing universalJavaApplicationStub from a build, renamed it to my app name 'PXPlay' and included it in the macConfig like that

customLauncher = file('PXPlay')

It seems it works that way and the CFBundleExecutable is now also PXPlay. Is this the intended way or is there any easier way? If not then does that have any disadvantages?

**Reply by fvarrui** [reply]
Hi @streamingdv!

This is the way right now. Maybe we can add a new MacConfig property to set a custom name for universalJavaApplicationStub (e.g. `macConfig.customName="PXPlay"`).

PRs are welcome 😃

**Comment by hugolaniewski-alt** [comment]
How can I hide an app against something like airlock enforcement agent on mac silicon by changing a cfbundle identifier without corrupting the app?
